.circleci/config.yml

```diff
@@ -0,0 +1,200 @@
+version: 2.1
+
+executors:
+  test-executor:
+    resource_class: xlarge
+    machine:
+      image: ubuntu-2004:202010-01
+      docker_layer_caching: true
+    environment:
+      IMAGE_NAME: chainsafe/pint
+      CARGO_AUDIT: /home/circleci/.cargo/bin/cargo-audit
+      RUSTC_WRAPPER: /home/circleci/.cargo/bin/sccache
+      RUSTC_TOOLCHAIN: nightly-2021-04-18
```

Comment:
What's best practice for selecting nightly toolchain. `lastest` can be incomplete sometimes, right? Is there any downside of setting this randomly to a recent date like we do here?

Reply:
Using `$ curl --proto '=https' --tlsv1.2 https://sh.rustup.rs -sSf | sh
` follows [installation](https://doc.rust-lang.org/book/ch01-01-installation.html#installation) with specified `nightly` channel will download the lastest nightly rust.

But I think using a specified version of rust can help us reduce the time modifying CI, substrate updates really fast, and sometimes the latest toolchains have bugs for compiling substrate.

For example, there was a big change in the rust compiler after `2020-10-05`, substrate could not be compiled with the latest toolchains for months, and rust toolchains after `2020-03-13`, have bugs compiling the debug info to wasm32-unknown-unknown target